MDANSE_GUI/Src/MDANSE_GUI/Tabs/Models/PlotDataModel.py

```diff
@@ -185,6 +204,42 @@ def add_file(self, filename: str):
             self._next_number += 1
             self.appendRow(new_item)
             new_item.populate(new_datafile._file)
+            self.finished_loading.emit(self._next_number - 1)
+            self.recent_files_update.emit(filename)
+
+    @Slot(str)
+    def recent_plot_files(self, plot_filepath: str):
+        """Adding recently loaded plot selection files to a json file. The json file is used to
+        populate the "Open Recent Plot Selection File" menu in the File menu.
+
+        Parameters
+        ----------
+        plot_filepath : Path of the added file
+        Returns
+        -------
+        Str
+            Recently loaded plot selection file path.
+        """
+        filename = self.DEFAULT_JSON_PATH
+        max_num_files = self.MAX_NUMBER_RECENT_FILES
+        if os.path.exists(filename) and os.path.getsize(filename) > 0:
+            with open(filename) as f:
+                recent_files = json.load(f)
+        else:
+            recent_files = []
+
+        if plot_filepath in recent_files:
+            recent_files.remove(plot_filepath)
+        recent_files.append(plot_filepath)
+
+        if len(recent_files) > max_num_files:
+            delete_number_of_files = len(recent_files) - max_num_files
+            while delete_number_of_files > 0:
+                recent_files.pop(0)
+                delete_number_of_files -= 1
```

Comment:
```suggestion
        recent_files = recent_files[:max_num_files]
```
No reason to have these loops also not sure why you're popping from the front? If you want that you may need to reverse the array or have some different logic.

However, in future, if you want to iterate over a range use `range`, `while` should only be used for an unknown length of loop:


```suggestion
        if len(recent_files) > max_num_files:
            delete_number_of_files = len(recent_files) - max_num_files
            for _ in range(delete_number_of_files):
                recent_files.pop(0)
```